cjworkbench/middleware/i18n.py

```diff
@@ -29,6 +33,8 @@ def __call__(self, request):
 
         # Code to be executed for each request/response after
         # the view is called.
+        if resolve(request.path_info).url_name != "set_locale":
```

Comment:
If you _don't_ add these two lines of code, which integration tests fail? If none of them do, let's nix this line. Otherwise, let's discuss.

I don't like hard-coding a special URL in this middleware. I feel it's jumping above its layer of abstraction.

Reply:
The issue was that the `locale` decided by this middleware is to be changed by the view.

Now, the middleware does not store this `locale` in the cookie; it stores `request.locale_id`, which has been changed by the view of `set_locale`. The thing to keep in mind is that this can be also done by other views.

Reply:
I think we're talking over each other here. I'll rephrase with a very specific question:

How does `request.locale_id` influence the logic in `/set_locale`?

Reply:
1. The first part of the middleware runs and looks for the current locale. Saves it in its `locale` local variable, `request.locale_id`, and django.
2. `set_locale` runs. Finds the new locale and saves it somewhere (currently, in `request.locale_id` and in user profile for registered users, formerly, in the cookie instead of `request.locale_id`).
3. The second part of the middleware runs. This must set a cookie. If it sets the cookie using its own `locale` local variable, it will undo (some of) the changes in `set_locale` view (that's why I excluded this url in the first version of the code). If it sets the cookie using `request.locale_id`, it will allow views to change the cookie indirectly.

Reply:
And if the middleware _doesn't_ set a cookie? What happens?

I would expect `set_locale` and login to set a cookie. I wouldn't expect middleware to set a cookie.